Jaden Taking Over for Peterson - Added improvements to the badge management page (DONE Jaden)

peterson337 opened the pull request:
# Description
This PR improves the Badge Management - Badge Development page by styling the filter inputs, making filter label text white in dark mode, toggling the filter button label between "Add Filters" and "Remove Filters", and fixing alignment of the filter section in dark mode.

Jaden took over from Peterson to resolve a Node version mismatch (branch was authored against Node 14, project is on Node 20), fix a SonarQube cognitive complexity violation (score 16, max 15) by extracting module-level helper functions, and fix the filter section alignment issue flagged by reviewers.

## Related PRs:
None

## Main changes explained:
- `BadgeDevelopment.jsx` - filter button label toggles between "Add Filters" / "Remove Filters"; filter label text uses `text-light`/`text-dark` for dark mode; filter rows use flexbox for proper alignment; filter logic extracted to module-level `badgeMatchesFilters` and `normalize` helpers to satisfy SonarQube complexity limit
- `BadgeDevelopment.module.css` - new CSS module for filter section layout and input styling (replaces inline styles)
- `BadgeManagement.jsx` - heading text uses `text-light`/`text-dark` for dark mode

## How to test:
1. Check out this branch
2. Run `npm install` then `npm start`
3. Clear site data/cache
4. Log in as an admin or owner user
5. Go to Other Links → Badge Management → Badge Development
6. In light mode: click "Add Filters" - verify inputs are styled and button label changes to "Remove Filters" on click
7. In dark mode: click "Add Filters" - verify filter label text is white and all filter rows are properly aligned

## Note:
None

https://github.com/user-attachments/assets/cf4b5394-b053-462f-899c-2dd859a590a1

peterson337 committed:
test

vaibhavkoladiya reviewed (approved):
Checked out the current branch, ran npm install, and cleared site data before logging in as an admin user. Navigated to Other Links → Badge Management → Badge Development, activated dark mode, and tested the "Add Filters" button. Verified that the text color is white and the button label updates correctly when toggled. Everything works as expected — approving the PR.

![Screenshot 2025-05-14 163806](https://github.com/user-attachments/assets/26d23d1a-6f18-430a-8c22-1653c9a26116)

Anusha-Gali reviewed (changes requested):
Hi Peterson,

I tried to review your PR locally and was unable to do so cause of node version discrepancy.

<img width="938" height="758" alt="Image" src="https://github.com/user-attachments/assets/eec26609-3a24-428b-962f-ee9216eb4ab5" />

peterson337 committed:
Merge branch 'development' into peterson-implement-badge-management-darkmode-filter-test

peterson337 committed:
peterson-implement-badge-management-darkmode-filter-test

peterson337 committed:
Update weeklySummariesAction.test.js

peterson337 committed:
peterson-implement-badge-management-darkmode-filter-test

HemanthNidamanuru reviewed (changes requested):
Hi Peterson, 

I tested this locally and the entire Badge Management filter section is not aligned properly in dark mode. The text color and button label changes need to be fixed for dark mode.

<img width="1881" height="1037" alt="Image" src="https://github.com/user-attachments/assets/bf74c9ba-14eb-4fea-a274-ae81158a881b" />

Jaden300 committed:
Fix filter section alignment in dark mode on Badge Development page

Convert filter rows from inline paragraph + input to flex rows with
align-items: center so labels and inputs sit on the same baseline
in both light and dark mode.

Jaden300 committed:
Merge remote-tracking branch 'origin/development' into peterson-implement-badge-management-darkmode-filter-test

Jaden300 committed:
Move filter styles to CSS module and fix cognitive complexity

Jaden300 committed:
Fix CSS Modules violations and reduce cognitive complexity in BadgeDevelopment

- Replace inline style objects with getBoxStyling(darkMode) and .btn CSS class
- Extract getDark helper to eliminate repeated darkMode ternaries
- Precompute labelClass and inputClass to reduce JSX branching

kzou55 reviewed (changes requested):
Hi Jaden,

I ran and tested the PR locally.

## Verified
-  Add Filters button present

  <img width="1887" height="374" alt="Image" src="https://github.com/user-attachments/assets/5535d68f-a304-40e3-a092-dd47cacc9093" />

- Clicking "Add Filters" causes the search, filter by type, and filter by rank inputs to apear and the Remove Filters button is present

  <img width="1902" height="546" alt="Image" src="https://github.com/user-attachments/assets/0b0bbb39-fe89-43b8-8ba8-a317a7686d9e" />

- "Add Filters" button looks goo in Dark Mode 

  <img width="1891" height="362" alt="Image" src="https://github.com/user-attachments/assets/da5f9c87-e5fb-4a0c-946e-aaa5243ff11e" />


## Issues
- The inputs and the text are not applied in Dark Mode consistently. 
  - The input field is white and doesn't have dark mode applied to it. 
    - Note: The text is white for some of them which is correct but the issue is that the input field doesn't have dark mode applied to them making it difficult to see
  - The drop down for "Filter by type" also has dark mode issues
  - The "choose a rank" input's placeholder text is not white(doesn't have dark mode applied to it) and the input field for it(white box) needs to have dark mode applied to it
    

  <img width="1909" height="539" alt="Image" src="https://github.com/user-attachments/assets/ae47ac16-ce27-4640-a087-3e06b72d77bb" />

  https://github.com/user-attachments/assets/a922c62d-1b93-4456-902f-c2612239315d

iAbhi001 reviewed (changes requested):
Hi @Jaden300,

I tested the PR locally in dark mode and verified that the "Add Filters" button is present and functioning properly. However, there are a few dark mode styling inconsistencies with the input fields that need to be fixed before merging:

Input Field Backgrounds: The text input boxes remain white in dark mode while the typed text turns white, making the text unreadable against the background. Please apply dark mode styling to the input container backgrounds.

"Filter by Type" Dropdown: The dropdown menu options are missing dark mode styling and have contrast issues.

"Choose a Rank" Placeholder: The placeholder text isn't adapting to dark mode and lacks proper contrast against the input field.

Once these contrast/theme fixes are in place, this should be ready to go. Thanks!

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/f193b35f-1ff6-4e1e-bfb4-9d04ddc506f4" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/834b84db-786a-48af-bf1e-51e16b935326" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/9e74c695-f3ce-4d2c-9155-6e10018d2eb4" />

DeepighaJ reviewed (changes requested):
- Tested the PR locally, on clicking the button Add filter changes to Remove Filter and filter input loads.
- The dark mode implementation needs improvement as they are not consistent across the inputs and the page.
- On selecting/highlighting a row goes white in dark mode, makes text invisible.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/c662d3ef-8f03-47d5-b32f-4f5d138e3ecd" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/7f386135-3e17-4bae-b5ba-6631f6484288" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/f1409f33-fdcc-482e-a4cc-0529ce6af986" />

Jaden300 committed:
Fix dark mode input styling and row selection on Badge Development page

- Remove inline styles from buttons; move box-shadow to CSS module
- Add ::placeholder rule so placeholder text inherits white color in dark mode
- Add ::selection rule to fix invisible text when highlighting rows in dark mode

iAbhi001 reviewed (dismissed):
### Local Testing & PR Review

I checked out the branch and tested the changes locally on Node 20.

**Verified:**
- [x] **Filter Toggle:** The "Add Filters" button correctly toggles to "Remove Filters" and displays the search/filter controls as expected.
- [x] **Dark Mode Input Styling:** Checked the dark mode implementation across input fields, placeholders, and the "Filter by type" dropdown. Dark mode backgrounds apply correctly with readable text contrast.
- [x] **Row Highlighting:** Verified that selected/highlighted rows keep high contrast and readable text in dark mode without turning white.
- [x] **Layout & Alignment:** The filter section flexbox alignment renders cleanly in both Light and Dark modes.

All checks pass and all previously requested dark mode contrast issues have been resolved. **LGTM! Approving the PR.**

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/0b576121-dbb7-4b4e-adf4-0c44cd864e98" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/63c8faae-0c06-4409-919b-288c5aad21af" />

<img width="1280" height="832" alt="Image" src="https://github.com/user-attachments/assets/265e2376-2c2c-4813-8575-a62517836429" />

DeepighaJ reviewed (changes requested):
Tested the PR locally, the dark mode issues are not fixed yet. The dropdown and hover still needs improvement.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/b1a86b5d-26ab-4156-9ab0-564a598645fb" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/40afb93f-46c4-4f99-9b48-054239997908" />

Jaden300 committed:
fix(dark-mode): fix filter input and dropdown dark mode styling on Badge Development page

kzou55 reviewed (commented):
Hi Jaden, 

I ran and tested the PR locally again. 

## Verified
- Previous behavior works correctly
  - Clicking "Add Filters" shows the inputs and "Remove Filters" button 
  -

kzou55 reviewed (changes requested):
Hi Jaden, 

I ran and tested the PR locally again. 

## Verified
- Dark Mode input fields and dropdown menu and text color are properly in dark mode and hover looks good

    <img width="1899" height="415" alt="Image" src="https://github.com/user-attachments/assets/f6ad7e4a-5d2e-4c7e-b9f3-0365d36aa58a" />

    <img width="1909" height="671" alt="Image" src="https://github.com/user-attachments/assets/9618dc93-4e72-45ac-8e5b-cb7b1fa97110" />
    
    <img width="1887" height="773" alt="image" src="https://github.com/user-attachments/assets/7abfe3a2-e9a8-405a-a079-b7f492e532d5" />



- Previous behavior works correctly
  - Clicking "Add Filters" shows the inputs and "Remove Filters" button 
  
    <img width="1894" height="394" alt="Image" src="https://github.com/user-attachments/assets/b7201889-d4a3-48ef-b229-24177efa45cb" />

    <img width="1895" height="460" alt="Image" src="https://github.com/user-attachments/assets/5aaf8361-a540-4a9a-b36a-a040c0ab4589" />

## Issues
1. The "No badges match the current filters" could use better contrast in Dark Mode

    <img width="646" height="426" alt="Image" src="https://github.com/user-attachments/assets/346aa140-18c2-4a2c-aff9-1fe4dd9dac61" />

2. Filter by Rank slider could have dark mode applied to it
    <img width="319" height="48" alt="Image" src="https://github.com/user-attachments/assets/22bdb6a6-cf8e-4383-b514-c6665c80e217" />

    <img width="322" height="54" alt="Image" src="https://github.com/user-attachments/assets/4b8e4b20-3a95-4b52-9501-36f315da6773" />

3. Could you clarify if this is within the PR spec, but I realized that Clicking "Remove Filters" hides the filter controls but leaves the previously applied filters active. This can be confusing because users may expect the filters to be removed or may not realize the results are still filtered. If the intended behavior is only to collapse the filter panel, consider renaming the button to "Hide Filters" or providing a visible indication that filters are still active. Otherwise, the button should clear the applied filters when it is clicked.

  https://github.com/user-attachments/assets/8adc0aa2-4af1-479a-a8a4-b6f15afa74ce

AdiDubbs commented on src/components/Badge/BadgeDevelopment.jsx:
```diff
@@ -19,64 +35,47 @@ function BadgeDevelopment(props) {
-        style={darkMode ? { ...boxStyleDark, margin: 20 } : { ...boxStyle, margin: 20 }}
       >
-        Add Filters
+        {isAddFiltersOpen ? 'Remove Filters' : 'Add Filters'}
```
The label toggles to "Remove Filters", but `toggleFilters` on line 36 only hides the panel - `searchName`, `searchType`, `rankFilter` and `chooseRankFilter` all keep their values, and `filteredBadgeData` still applies them. So after typing a badge name and clicking "Remove Filters", the list stays filtered with no visible input explaining why it's short, and the only way to clear it is to reopen the panel and empty each field.

Either resetting the four filter states in `toggleFilters`, or naming the button "Hide Filters", would match what it actually does.

AdiDubbs commented on src/actions/__tests__/weeklySummariesAction.test.js:
```diff
@@ -70,15 +67,16 @@ describe('Weekly Summaries', () => {
+      };
 
+      axios.get.mockResolvedValue({ data: mockData, status: 200 });
       axios.get.mockResolvedValue({ data: mockData, status: 200 });
```
`axios.get.mockResolvedValue(...)` is duplicated on lines 79 and 80 - it looks like it slipped in during the reformat. It's probably harmless since the second call is identical, but worth removing.

More generally, this file and `weeklySummariesReportActions.test.js` are only being reformatted here (quotes and trailing commas) and aren't related to badge management. I checked and no coverage is lost - 8 tests and the same assertion counts in both files before and after - but it's about 110 lines of the diff that reviewers have to read past. Might be cleaner as a separate formatting PR.

AdiDubbs reviewed (changes requested):
This is nicely done overall. I checked the extracted `badgeMatchesFilters` against the original inline logic branch by branch and it's equivalent, all ten `styles[...]` lookups resolve, and `.btn`/`.btnDark` reproduce `boxStyle` exactly. Worth calling out that the new dark-mode table styling in `BadgeDevelopmentTable.module.css` is properly scoped through `.tableDark` — dropping `dark-mode` from the table element is safe too, since ThemeManager puts that class on `body` anyway.

One behaviour issue is that the button now says "Remove Filters" but only hides the panel. The filters stay applied to the badge list, so you can end up looking at a filtered table with no visible reason for it.

There are two smaller things to note as well: There's a duplicated `axios.get.mockResolvedValue` line in weeklySummariesAction.test.js, and the two weeklySummaries test files are only reformatted here - no coverage lost, but unrelated to badges.

It's also worth coordinating with #3631, which is yours too and touches BadgeDevelopmentTable.jsx as well. Both PRs add dark-mode hover styling for the same table - this one scopes it through a local class, that one applies it globally to every table in the app. They don't conflict textually but they'd both be live, so it's probably worth doing it once, and the way it's done here is probably better.

The branch is 618 commits behind development, so it'll also need a rebase before this can be properly tested.

Jaden300 committed:
fix: rename Remove Filters to Hide Filters, fix dark mode contrast and slider styling, remove duplicate mock line

AdiDubbs reviewed (approved):
Both fixed, thank you. "Hide Filters" is the honest name given it only toggles the panel, and the duplicate mock line is gone. The dark-mode additions on the slider and the empty-state message look right too. I'll be approving.

kzou55 reviewed (approved):
Hi Jaden, 

I ran and tested the PR locally again. 

## Verified
- Click "Add Filters" - verify inputs are styled and button label changes to "Remove Filters" on click

  <img width="1909" height="990" alt="Image" src="https://github.com/user-attachments/assets/afe35943-df3b-4294-930a-29985c434c71" />

- In dark mode: click "Add Filters" - verify filter label text is white and all filter rows are properly aligned

  <img width="1896" height="979" alt="Image" src="https://github.com/user-attachments/assets/b2a621be-33fe-4d85-a4c5-bd58f56b975c" />

- Improvements confirmed:
  - Filter button is now "Hide Filters"
  - Input slider has improved dark mode 
  - No badge found message in the correct color
    <img width="1900" height="432" alt="Image" src="https://github.com/user-attachments/assets/2c78a73a-c913-4c10-adca-2fe7e5d4a584" />
